feat(surveys): add response length validation for open text questions

ordehi opened the pull request:
Adds the ability to configure min/max character limits for open text survey questions. The validation is performed client-side by the SDK.

Depends on https://github.com/PostHog/posthog-js/pull/2820

## Changes

<img width="1536" height="1210" alt="CleanShot 2025-12-29 at 17 57 35@2x" src="https://github.com/user-attachments/assets/6ab461fc-c944-484c-b965-b07a3d594f41" />


- Add SurveyValidationType enum and SurveyValidationRule interface
- Add ValidationRulesEditor component with simple checkbox UI
- Add validation field to BasicSurveyQuestion type
- Add backend validation for validation rules structure
- Add SDK version requirement entry (placeholder version)
- Add 5 backend tests for validation rules

## How did you test this code?

- Wrote tests, tested the UI locally

Other tests I should run:

- [x] Create a survey with an Open text question
- [x] Enable "Message length" validation with min/max values
- [x] Save survey and verify validation persists
- [x] Backend tests pass: `pytest posthog/api/test/test_survey.py -k validation_rules`"
- [ ] Use the survey with the SDK locally

## Publish to changelog?

<!-- For features only. Check the box if this should go in our company changelog. -->
<!-- Removing this section or leaving it blank means this feature will probably remain unpublished. -->

- [x] 🚀 Yes, publish to #changelog

<!-- If publishing, your PR must include a 1-3 sentence description of the feature plus a screenshot or screen recording if applicable. 🙏 -->

<!-- Check #changelog in Slack for more details. -->

ordehi committed:
feat(surveys): add response length validation for open text questions

Adds the ability to configure min/max character limits for open text
survey questions. The validation is performed client-side by the SDK.

- Add SurveyValidationType enum and SurveyValidationRule interface
- Add ValidationRulesEditor component with simple checkbox UI
- Add validation field to BasicSurveyQuestion type
- Add backend validation for validation rules structure
- Add SDK version requirement entry (placeholder version)
- Add 5 backend tests for validation rules

ordehi committed:
refactor(surveys): simplify validation UI and fix spacing

- Always show min/max inputs (disabled when unchecked) to prevent layout shift
- Use checkbox label pattern matching the Optional checkbox
- Remove custom margins, rely on parent gap-2 for consistent spacing

ordehi committed:
fix(surveys): add min/max cross-validation for message length

- Frontend: constrain min input to not exceed max, and vice versa
- Backend: reject validation rules where min_length > max_length
- Add test for the cross-validation error case

ordehi committed:
refactor(surveys): use explicit undefined check for clearer intent

ordehi committed:
fix(surveys): improve validation rules UX and remove unused Email type

- Remove disabled state from min/max inputs so typing auto-enables validation
- Follow existing "Delay survey popup" pattern for checkbox behavior
- Remove unused Email enum from SurveyValidationType

ordehi committed:
fix(surveys): simplify validation input value props

adboio commented on frontend/src/scenes/surveys/surveyVersionRequirements.ts:
```diff
@@ -128,6 +129,18 @@ export const SURVEY_SDK_REQUIREMENTS: SurveyFeatureRequirement[] = [
     },
+    {
+        feature: 'Response length validation',
+        sdkVersions: { 'posthog-js': 'X.X.X' },
```
i think your sdk changes covered react-native too, just remember to add that here before merging :saluting_face:

adboio reviewed (approved):
lgtm! just get that SDK warning updated and :ship::ship:

ordehi committed:
chore(surveys): update SDK version requirements for response validation

github-actions[bot] committed:
test(storybook): update UI snapshots

ordehi committed:
Merge branch 'master' into feat/survey-validation-ui

ordehi committed:
fix(surveys): add posthog_flutter to Response length validation unsupportedSdks

The test requires all SDK types to be covered for each feature.
posthog_flutter was missing from the Response length validation entry.

github-actions[bot] committed:
test(storybook): update UI snapshots

ordehi committed:
Merge branch 'master' into feat/survey-validation-ui

ordehi committed:
fix(surveys): require positive integers for validation rule values

ordehi committed:
Merge branch 'master' into feat/survey-validation-ui

github-actions[bot] committed:
test(storybook): update UI snapshots

ordehi committed:
Merge branch 'master' into feat/survey-validation-ui
